modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NancyFXServerCodegen.java

```diff
@@ -0,0 +1,404 @@
+            parentModel.hasChildren = true;
+            final Collection<CodegenModel> childrenModels = childrenByParent.get(parent);
+            for (final CodegenModel child : childrenModels) {
+                processParentPropertiesInChildModel(parentModel, child);
+            }
+        }
+    }
+
+    private CodegenModel modelByName(final String name, final Map<String, Object> models) {
+        final Object data = models.get(name);
+        if (data instanceof Map) {
+            final Map<?, ?> dataMap = (Map<?, ?>) data;
+            final Object dataModels = dataMap.get("models");
+            if (dataModels instanceof List) {
+                final List<?> dataModelsList = (List<?>) dataModels;
+                for (final Object entry : dataModelsList) {
+                    if (entry instanceof Map) {
+                        final Map<?, ?> entryMap = (Map<?, ?>) entry;
+                        final Object model = entryMap.get("model");
+                        if (model instanceof CodegenModel) {
+                            return (CodegenModel) model;
+                        }
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private void processParentPropertiesInChildModel(final CodegenModel parent, final CodegenModel child) {
+        final Map<String, CodegenProperty> childPropertiesByName = new HashMap<>(child.vars.size());
+        for (final CodegenProperty property : child.vars) {
+            childPropertiesByName.put(property.name, property);
+        }
+        CodegenProperty previousParentVar = null;
+        for (final CodegenProperty property : parent.vars) {
+            final CodegenProperty duplicatedByParent = childPropertiesByName.get(property.name);
+            if (duplicatedByParent != null) {
+                log.info(String.format("Property: '%s' in '%s' model is inherited from '%s'" ,
+                        property.name, child.classname, parent.classname));
+                duplicatedByParent.isInherited = true;
+                final CodegenProperty parentVar = duplicatedByParent.clone();
+                parentVar.hasMore = false;
+                child.parentVars.add(parentVar);
+                if (previousParentVar != null) {
+                    previousParentVar.hasMore = true;
+                }
+                previousParentVar = parentVar;
+            }
+        }
+    }
+
+    @Override
+    public void postProcessModelProperty(final CodegenModel model, final CodegenProperty property) {
+        super.postProcessModelProperty(model, property);
+        if (!isNullOrEmpty(model.parent)) {
+            parentModels.add(model.parent);
+            if (!childrenByParent.containsEntry(model.parent, model)) {
+                childrenByParent.put(model.parent, model);
+            }
+        }
+    }
+
+    @Override
+    public String toEnumVarName(final String name, final String datatype) {
```

Comment:
I would think the logic here could be moved up to AbstractCSharpCodegen, because Enum names shouldn't be converted differently for C# depending on which generator is used.